datalake/src/main/java/com/sequenceiq/datalake/flow/delete/SdxDeleteActions.java

```diff
@@ -189,6 +190,13 @@ protected void doExecute(SdxContext context, SdxDeletionFailedEvent payload, Map
                             payload.getResourceId());
                     metricService.incrementMetricCounter(MetricType.SDX_DELETION_FAILED, sdxCluster);
                     eventSenderService.notifyEvent(context, ResourceEvent.SDX_CLUSTER_DELETION_FAILED);
+
+                    if (sdxCluster.isDetached()) {
+                        eventSenderService.sendEventAndNotification(
```

Comment:
What's the difference between `eventSenderService.sendEventAndNotification` and the ` eventSenderService.notifyEvent` call right above this?

Reply:
There is no real difference, `notifyEvent` internally calls `sendEventAndNotification`.
`sendEventAndNotification` allows for message arguments to be passed in as well though, which is needed for the string formatting used by the events I am sending.

Reply:
Basically, if you have an event with a string that needs to be formatted you use the `sendEvent` one, otherwise `notifyEvent` suffices.

Not really sure why the two exist though.

Reply:
I'm wondering if we should have an if/else here instead of just the if. Like instead of always sending `eventSenderService.notifyEvent(context, ResourceEvent.SDX_CLUSTER_DELETION_FAILED);`, we only do it in an `else` if it's not detached. So we're sending one event of the other, not both. But I don't know enough about this code to know if we need to always send the SDX_CLUSTER_DELETION_FAILED event, so I'll leave that to your discretion.

Reply:
Done I think.